File: 0776-n-ary-tree-postorder-traversal/0776-n-ary-tree-postorder-traversal.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution:
    def postorder(self, root: 'Node') -> List[int]:
        result = []
        if not root:
            return result
        
        stack = [root]
        while stack:
            current = stack.pop()
            result.append(current.val)
            for child in current.children:
                stack.append(child)
        result.reverse()
        return result

File: 0776-n-ary-tree-postorder-traversal/0776-n-ary-tree-postorder-traversal.java
/*
// Definition for a Node.
class Node {
    public int val;
    public List<Node> children;

    public Node() {}

    public Node(int _val) {
        val = _val;
    }

    public Node(int _val, List<Node> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
    public List<Integer> postorder(Node root) {
        Stack<Node> stack = new Stack<>();
        List<Integer> result = new ArrayList<>();
        if(root == null){
            return result;
        }
       stack.push(root);
        while(stack.size() > 0){
            Node current = stack.pop();
            result.add(current.val);
            for(Node child: current.children){
                stack.push(child);
            }
        }
        Collections.reverse(result);
        return result;
    }
}